platform/api.scripting/src/org/netbeans/api/scripting/Scripting.java

```diff
@@ -0,0 +1,457 @@
+ * to a Java {@link java.lang.Number} and then to a Java {@code int}.</li>
+ * <li>Parentheses around the JS function definition keep it out of JavaScript's
+ * global scope, so the Java object holds the only reference to it.</li>
+ * </ul>
+ * <p>
+ * The multiple steps needed to convert the result in the above example
+ * produces awkward code that can be simplified.
+ * Instead of invoking the JS function directly, and "casting" the wrapped JS result,
+ * we can instead
+ * {@linkplain javax.script.Invocable#getInterface "cast"}
+ * the JS function to a Java foreign function (of type {@code MomentConverter}) that
+ * returns the desired Java type directly, as shown in the following variation.
+ * <p>
+ * {@codesnippet org.netbeans.api.scripting.ScriptingTutorial#accessFieldsOfJavaObjectWithConverter}
+ *
+ * <h3>Access Java constructors and static methods from JavaScript</h3>
+ *
+ * Dynamic languages can also access the <em>public constructors</em> and <em>public static methods</em>
+ * of any Java class for which they are provided a reference.
+ * The following example shows JavaScript access to the public constructor of a Java
+ * class.
+ * <p>
+ * {@codesnippet org.netbeans.api.scripting.ScriptingTutorial#createJavaScriptFactoryForJavaClass}
+ *
+ * Notes:
+ * <ul>
+ * <li>Evaluating the JS source returns an anonymous JS function of one argument
+ * assigned to {@code jsFunction} that can be executed directly with one argument.</li>
+ * <li>The Java class argument {@code Moment.class} is seen by the JS function as a
+ * <em>foreign class</em> whose public constructor is visible.</em>
+ * <li>Executing {@code jsFunction} with the Java class argument returns
+ * a JS "factory" function (for the Java class) that can be
+ * {@linkplain javax.script.Invocable#getInterface "cast"}
+ * to the desired Java function type ({@code MomentFactory}).</li>
+ * <li>Parentheses around the JS function definition keep it out of JavaScript's
+ * global scope, so the Java object holds the only reference to it.</li>
+ * </ul>
+ *
+ * </div>
+ * <script src="doc-files/tutorial.js"></script>
+ *
+ * @since 1.0
+ */
+public final class Scripting {
+    private Scripting() {
+    }
+
+    /** Create new {@link ScriptEngineManager} configured for the NetBeans
+     * environment. The manager serves as an <em>isolated</em> environment -
+     * engines created from the same manager are supposed to share the
+     * same internals and be able to communicate with each other.
+     *
+     * @return new instance of the engine manager
+     */
+    public static ScriptEngineManager createManager() {
+        List<ScriptEngineFactory> extra = new ArrayList<>();
+        for (EngineProvider p : Lookup.getDefault().lookupAll(EngineProvider.class)) {
+            extra.addAll(p.factories());
+        }
+        ClassLoader l = Lookup.getDefault().lookup(ClassLoader.class);
+        if (l == null) {
+            l = Thread.currentThread().getContextClassLoader();
```

Comment:
Just a thought: `contextClassLoader` is sometimes used to override defaults; should it be 2nd in the line ?

Reply:
This is the standard code and order spread on many places.